tools/src/test/java/org/apache/kafka/tools/ConsoleShareConsumerOptionsTest.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.tools;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.utils.Exit;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class ConsoleShareConsumerOptionsTest {
+
+    @Test
+    public void shouldParseValidConsumerValidConfig() throws IOException {
+        String[] args = new String[]{
+                "--bootstrap-server", "localhost:9092",
+                "--topic", "test",
+        };
+
+        ConsoleShareConsumerOptions config = new ConsoleShareConsumerOptions(args);
+
+        assertEquals("localhost:9092", config.bootstrapServer());
+        assertEquals("test", config.topicArg());
+        assertFalse(config.rejectMessageOnError());
+        assertEquals(-1, config.maxMessages());
+        assertEquals(-1, config.timeoutMs());
+    }
+
+    @Test
+    public void shouldExitOnUnrecognizedNewConsumerOption() {
+        Exit.setExitProcedure((code, message) -> {
+            throw new IllegalArgumentException(message);
+        });
+
+        String[] args = new String[]{
+                "--new-consumer",
+                "--bootstrap-server", "localhost:9092",
+                "--topic", "test"
+        };
+
+        try {
+            assertThrows(IllegalArgumentException.class, () -> new ConsoleShareConsumerOptions(args));
+        } finally {
+            Exit.resetExitProcedure();
+        }
+    }
+
+    @Test
+    public void shouldParseValidConsumerConfigWithAutoOffsetResetLatest() throws IOException {
+        String[] args = new String[]{
+                "--bootstrap-server", "localhost:9092",
+                "--topic", "test",
+                "--consumer-property", "auto.offset.reset=latest"
```

Comment:
`auto.offset.reset` is not supported so these tests don't make sense. Maybe use `session.timeout.ms` as the configuration for these tests.